src/systems/a320_systems/src/flaps_computer.rs

```diff
@@ -0,0 +1,1347 @@
+    }
+
+    fn set_target_slats_angle(&mut self, angle: Angle) {
+        self.slats_demanded_angle = angle;
+    }
+
+    fn target_flaps_angle_from_state(flap_state: FlapsConf) -> Angle {
+        match flap_state {
+            FlapsConf::Conf0 => Angle::new::<degree>(0.),
+            FlapsConf::Conf1 => Angle::new::<degree>(0.),
+            FlapsConf::Conf1F => Angle::new::<degree>(10.),
+            FlapsConf::Conf2 => Angle::new::<degree>(15.),
+            FlapsConf::Conf3 => Angle::new::<degree>(20.),
+            FlapsConf::ConfFull => Angle::new::<degree>(40.),
+        }
+    }
+
+    fn target_slats_angle_from_state(flap_state: FlapsConf) -> Angle {
+        match flap_state {
+            FlapsConf::Conf0 => Angle::new::<degree>(0.),
+            FlapsConf::Conf1 => Angle::new::<degree>(18.),
+            FlapsConf::Conf1F => Angle::new::<degree>(18.),
+            FlapsConf::Conf2 => Angle::new::<degree>(22.),
+            FlapsConf::Conf3 => Angle::new::<degree>(22.),
+            FlapsConf::ConfFull => Angle::new::<degree>(27.),
+        }
+    }
+
+    fn generate_configuration(
+        &self,
+        handle_transition: Option<(usize, usize)>,
+    ) -> Option<FlapsConf> {
+        if let Some((from, to)) = handle_transition {
+            match from {
+                0 => match to {
+                    1 => {
+                        if self.indicated_airspeed.get::<knot>()
+                            <= Self::HANDLE_ONE_CONF_AIRSPEED_THRESHOLD_KNOTS
+                        {
+                            Some(FlapsConf::Conf1F)
+                        } else {
+                            Some(FlapsConf::Conf1)
+                        }
+                    }
+                    0 => Some(FlapsConf::from(0)),
+                    _ => Some(FlapsConf::from(to + 1)),
+                },
+                1 => match to {
+                    1 => {
+                        if self.indicated_airspeed.get::<knot>()
+                            > Self::CONF1F_TO_CONF1_AIRSPEED_THRESHOLD_KNOTS
+                        {
+                            Some(FlapsConf::Conf1)
+                        } else {
+                            None
+                        }
+                    }
+                    0 => Some(FlapsConf::from(0)),
+                    _ => Some(FlapsConf::from(to + 1)),
+                },
+                _ => match to {
+                    1 => {
+                        if self.indicated_airspeed.get::<knot>()
+                            <= Self::CONF1F_TO_CONF1_AIRSPEED_THRESHOLD_KNOTS
+                        {
+                            Some(FlapsConf::Conf1F)
+                        } else {
+                            Some(FlapsConf::Conf1)
+                        }
+                    }
+                    0 => Some(FlapsConf::from(0)),
+                    _ => Some(FlapsConf::from(to + 1)),
+                },
+            }
+        } else {
+            None
+        }
```

Comment:
**suggestion**: The matching can be simplified.

```suggestion
        match handle_transition {
            Some((0, 1))
                if self.indicated_airspeed.get::<knot>()
                    <= Self::HANDLE_ONE_CONF_AIRSPEED_THRESHOLD_KNOTS =>
            {
                Some(FlapsConf::Conf1F)
            }
            Some((0, 1)) => Some(FlapsConf::Conf1),
            Some((1, 1))
                if self.indicated_airspeed.get::<knot>()
                    > Self::CONF1F_TO_CONF1_AIRSPEED_THRESHOLD_KNOTS =>
            {
                Some(FlapsConf::Conf1)
            }
            Some((1, 1)) => None,
            Some((_, 1))
                if self.indicated_airspeed.get::<knot>()
                    <= Self::CONF1F_TO_CONF1_AIRSPEED_THRESHOLD_KNOTS =>
            {
                Some(FlapsConf::Conf1F)
            }
            Some((_, 1)) => Some(FlapsConf::Conf1),
            Some((_, 0)) => Some(FlapsConf::from(0)),
            Some((_, to)) => Some(FlapsConf::from(to + 1)),
            None => None,
        }
```

Reply:
I am not sure this is equivalent. This is again problematic with what we talked about in Discord regarding handle at position `1`. When the transition `0->1` occurs, e.g. when KIAS > 100 but smaller than 210, then the condition in line 133 is matched. In the next tick the transition is `1->1`, and the condition in line 141 will be matched, which will wrongly yield `Conf1F`.

Reply:
Implemented in `5b14499 `